### ISSUE TITLE
Need a way to access the current title of loaded web content

### ISSUE THREAD

**Issue by lloyd** [mozilla/chromeless#35]
API for explicit access of title could be similar to other iframe controls (as discussed in Issue #34).  We might also augment events emitted which convey iframe load percentage so clients can get at the title immediately as it changes.

**Comment by taboca** [comment]
Created commit SHA: a1f658d initial input to this but in fact this is related to another design aspect. 

As Lloyd says, if we are to put iframe.stop() and iframe.favIconURL or iframe.title then we could consider using XBL. The XBL could be also means to activate the whole iframe binding. So then it would be perhaps a new tag, such as 

<webframe /> 

that has an inner <iframe /> in it and does all the magic.

**Comment by taboca** [comment]
SHA: a1f658d6cd2dac1df5fd5ad055f550dc457028f1

**Comment by taboca** [comment]
The approach to do here, on demand, is to get the DOMTitleChanged event:

```
iframeElement.addEventListener("DOMTitleChanged", function (e) {
     console.log("!!!!  title is " + e.target.title);
}, false);
```

Ref: https://developer.mozilla.org/En/Listening_to_events

**Comment by lloyd** [comment]
beautiful, marcio

**Comment by taboca** [comment]
Note pause --> this direction seems to be right solution for us to architecture FavIcon availability, the feeds availability, and perhaps other things. I believe that a developer's scenario here is someone willing to be notified about all these things ( that can be on demand ) for a given iframe or collection of iframes. 

I initiated some experiments to make a XBL-based browser tag, that has the iframe and everything in it ( events are towards it, so securely exposed to developer ) all the internal binding as part of its constructor: 

https://github.com/taboca/chromeless/commit/438fee399a9ab6b7e3e65bf1e6681cd1cc3d1e62

While it shows some direction I also got some problems. For example, the event handler code cannot see the local JS scope, for example call alert function. Not sure exactly if I am using XBL wrong. Other thing is that it did not really activated XBL with the page normal privilege.

So maybe the simplest, for now, thing to do is to fire again another SuperPowersTitleChanged event against our iframe element.
